backend/app/database/client.py

```diff
@@ -0,0 +1,96 @@
+from typing import cast
+
+from postgrest import APIResponse, SyncFilterRequestBuilder, SyncSelectRequestBuilder
+from supabase import Client, create_client
+
+from app.models import settings
+
+
+class DatabaseClient:
+    """A Database client class."""
+
+    def __init__(self) -> None:
+        self.__database_client: Client | None = self.__select_database_client()
```

Comment:
The nulls everywhere in this class are a code smell. The design you probably want here is a high-level state interface ("get user state" and not "execute this SQL query") that can be implemented against an external database or as an in-memory instance (or mocks or ...).